src/server/game/Scripting/ScriptMgr.cpp

```diff
@@ -96,6 +96,10 @@ template<>
 struct is_script_database_bound<AchievementCriteriaScript>
     : std::true_type { };
 
+template<>
+struct is_script_database_bound<AreaTriggerEntityScript>
```

Comment:
Yep, good catch. it has to be done for every database bound script (that scripts which use scriptnames in the database).